src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -461,18 +461,95 @@ private void ProcessRenderQueue()
             }
         }
 
-        private Task InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents)
+        private Task InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents, Task updateDisplayTask)
```

Comment:
I'm getting a little concerned the pattern of firing of tasks and then trying to reason about what happens to them. For instance this method is private and task-returning, and deals with the result of another task, but I'm not sure how it could have unit tests because it's private and used in a fire-and-forget way.


You and @SteveSandersonMS are more familar with the renderer than I, but I don't feel like I can give any meaningful feedback other than to say that this is getting pretty complicated and has history of having subtle bugs.

Reply:
It's fair, but we can actually write ok unit tests for this with a bit of help from mocks. I wrote a couple of them for this scenario specifically.